### PR TITLE
refactoring of `moving_window_calcs_2`, add README, contributors, expand tests

### PR DESCRIPTION
**Pull request by dfalster**
Hi @wcornwell , Great work with `moving_window_calcs`

While it was awesome, further optimisation was possible by boring everything into one function. Otherwise we end up recalculating things like meanx (and y and z) a bunch of times. 

Check this out? if you're happy, we can update to this new version. 

Comparing speeds:

**Master, using Zoo for skewness and kurtosis**

```
> system.time({ moving_window_calcs(df)})
    user   system  elapsed 
 828.672  167.050 1037.330 
```

**After replacing roll_skewness for something faster, and dropping kurtosis**

```
> system.time({moving_window_calcs(df)})
   user  system elapsed 
70.618   7.767  78.680 
```

**Brining all into one function, to reduce any duplication**

```
> system.time({new2 <- moving_window_calcs2(df)})
   user  system elapsed 
 41.320   5.026  47.420 
```

**Comparing all three methods**

```
microbenchmark::microbenchmark(
  doAccloop_all(data),
  moving_window_calcs(data),
  moving_window_calcs_2(data)
)
#> Warning in microbenchmark::microbenchmark(doAccloop_all(data), moving_window_calcs(data), : less accurate nanosecond times to avoid potential integer overflows
#> Unit: milliseconds
#>                         expr         min          lq        mean      median          uq        max neval cld
#>          doAccloop_all(data) 1874.630536 1924.027336 1957.852846 1944.725366 1975.835469 2156.93837   100  a 
#>    moving_window_calcs(data)    8.334029    8.542699    8.890808    8.678962    8.841568   12.56634   100   b
#>  moving_window_calcs_2(data)    7.946661    8.187188    8.807385    8.397169    8.768752   12.94694   100   b
```

### CODE REVIEW

**Review by wcornwell — APPROVED** [review]

**Comment by wcornwell** [comment]
Harder to test, but should be faster!

File: DESCRIPTION [hunk 1]
```diff
@@ -12,11 +15,10 @@ Imports:
     arrow,
     data.table,
     RcppRoll,
-    lubridate,
-    zoo,
-    e1071
```
Comment (wcornwell) [inline DESCRIPTION:17]: Probably still need this for testing?

File: R/internal.R [hunk 2]
```diff
@@ -74,7 +82,7 @@ rolling_mean_time_date <- function(date_time_vec,window_size) {
   # Convert to numeric
   numeric_dates <- as.numeric(lubridate::dmy_hms(date_time_vec))
   # Calculate the mean
-  mean_numeric <- RcppRoll::roll_mean(numeric_dates,
+  mean_numeric <- RcppRoll::roll_var(numeric_dates,
```
Comment (wcornwell) [inline R/internal.R]: I don't get this, but maybe I misunderstood original purpose

File: R/sliding_window.R [hunk 3]
```diff
@@ -121,14 +122,101 @@ moving_window_calcs <- function(df, window_size=50) {
...
+   
+  # ??? what do we call these
+  ODBA = abs_x + abs_y + abs_z
+  VDBA = sqrt(x_2 + y_2 + z_2)
+
+  out <- 
+    dplyr::tibble(
+      time = rolling_mean_time_date(df$Timestamp, window_size)
+    ) %>% 
+    dplyr::mutate(
```
Comment (wcornwell) [inline R/sliding_window.R:182]: Is this fastest way for big columns?

**Review by wcornwell — APPROVED** [review]

File: DESCRIPTION (hunk 1 above)
Comment (dfalster) [reply DESCRIPTION:17]: No, I brought in a copy of skewness package for testing purposes. Otherwise not needed

File: R/sliding_window.R (hunk 3 above)
Comment (dfalster) [reply R/sliding_window.R:182]: Not sure. But it's going pretty fast!

File: R/internal.R (hunk 2 above)
Comment (dfalster) [reply R/internal.R]: Good catch, this is a mistake I and will revert!

**Pull request merged**